pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -889,50 +890,67 @@ func (c *controller) drainNode(deleteMachineRequest *driver.DeleteMachineRequest
 			)
 		}
 
-		buf := bytes.NewBuffer([]byte{})
-		errBuf := bytes.NewBuffer([]byte{})
-
-		drainOptions := drain.NewDrainOptions(
-			c.targetCoreClient,
-			timeOutDuration,
-			maxEvictRetries,
-			pvDetachTimeOut,
-			nodeName,
-			-1,
-			forceDeletePods,
-			true,
-			true,
-			true,
-			buf,
-			errBuf,
-			c.driver,
-			c.pvcLister,
-			c.pvLister,
-		)
-		err = drainOptions.RunDrain()
-		if err == nil {
-			// Drain successful
-			klog.V(2).Infof("Drain successful for machine %q. \nBuf:%v \nErrBuf:%v", machine.Name, buf, errBuf)
+		// update node with the machine's state prior to termination
+		if err = c.UpdateNodeTerminationCondition(machine); err != nil {
+			if forceDeleteMachine {
+				klog.Warningf("Failed to update node conditions: %v. However, since it's a force deletion shall continue deletion of VM.", err)
+			} else {
+				klog.Errorf("Drain failed due to failure in update of node conditions: %v", err)
```

Comment:
Should we return error on failure here ?
- Original PR https://github.com/gardener/machine-controller-manager/pull/492/files#diff-d8287fe74b5273163c9f6b6c635ad912R609

Reply:
I have avoided returning error here to make sure that the machine object status is updated on failure here - https://github.com/gardener/machine-controller-manager/blob/e073a81ab725d699a89044876ce8c1bfbb2eff33/pkg/util/provider/machinecontroller/machine_util.go#L900-L904.

At the end of the function the error is still returned here - https://github.com/gardener/machine-controller-manager/blob/e073a81ab725d699a89044876ce8c1bfbb2eff33/pkg/util/provider/machinecontroller/machine_util.go#L975